### PR TITLE
Improve flowgraph xml/dot dumping

### LINKED ISSUE #43712: Decouple flow graph dumps from JIT dumps
Using `COMPlus_JitDumpFg` requires also setting `COMPlus_JitDump`. It should be decoupled so you can choose to just dump the flow graph without also getting the dump.

(Yes, you'll very likely also want the dump, but it doesn't need to be a prerequisite.)

Also, `Compiler::fgOpenFlowGraphFile()` parses the `COMPlus_JitDumpFg` string for class/method name match. It seems like this code can all be deleted if JitDumpFg is changed to a `CONFIG_METHODSET` type, and the built-in `.contains()` function is used.

category:implementation
theme:debug-dumps
skill-level:beginner
cost:small

### PR DESCRIPTION
**Pull request by BruceForstall**
1. Decouple it from JitDump
2. Add some documentation, including a useful link to https://sketchviz.com/
3. Fix DUMP_FLOWGRAPHS Release build
4. Fix some xml dumping issues
5. Replace custom function pattern parsing with CONFIG_METHODSET ".contains" function
6. Make dot format the default (instead of xml)

Fixes #43712

### CODE REVIEW

**Comment by BruceForstall** [comment]
@AndyAyersMS @dotnet/jit-contrib PTAL

File: src/coreclr/jit/fgdiagnostic.cpp [hunk 1]
```diff
@@ -645,44 +580,49 @@ FILE* Compiler::fgOpenFlowGraphFile(bool* wbDontClose, Phases phase, LPCWSTR typ
...
+//      - The command:
+//           "C:\Program Files (x86)\Graphviz2.38\bin\dot.exe" -Tsvg -oFoo.svg -Kdot Foo.dot
+//        will produce a Foo.svg file that can be opened with any svg-capable browser.
+//    - https://sketchviz.com/
 //    - http://rise4fun.com/Agl/
 //      - Cut and paste the graph from your .dot file, replacing the digraph on the page, and then click the play
 //        button.
 //      - It will show a rotating '/' and then render the graph in the browser.
-//    MSAGL has also been open-sourced to https://github.com/Microsoft/automatic-graph-layout.git.
+//    MSAGL has also been open-sourced to https://github.com/Microsoft/automatic-graph-layout.
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgdiagnostic.cpp:591]: Last I looked MSAGL no longer included a dot parser.

File: src/coreclr/jit/fgdiagnostic.cpp [hunk 2]
```diff
@@ -645,44 +580,49 @@ FILE* Compiler::fgOpenFlowGraphFile(bool* wbDontClose, Phases phase, LPCWSTR typ
...
 
+    FILE* fgxFile = fgOpenFlowGraphFile(&dontClose, phase, createDotFile ? W("dot") : W("fgx"));
     if (fgxFile == nullptr)
     {
         return false;
     }
+
     bool        validWeights  = fgHaveValidEdgeWeights;
-    double      weightDivisor = (double)fgCalledCount;
+    double      weightDivisor = (double)(fgCalledCount == BB_ZERO_WEIGHT ? 1.0 : fgCalledCount); // avoid divide by zero
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgdiagnostic.cpp:625]:
```suggestion
    double      weightDivisor = (double)fgFirstBB->getCalledCount(this);
```
but maybe make that method static and / or move off of BasicBlock.

**Review by AndyAyersMS — APPROVED** [review]
Just a couple minor suggestions; feel free to disregard.

File: src/coreclr/jit/fgdiagnostic.cpp (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/jit/fgdiagnostic.cpp:591]: Interesting. Well, it's still useful to have a link.

**Pull request merged**